services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -415,38 +468,40 @@ public void recordIncomingRequest() {
     totalInflightRequestSensor.record();
   }
 
-  Attributes getRequestMetricDimensions(
-      HttpResponseStatus responseStatus,
-      VeniceResponseStatusCategory veniceResponseStatusCategory) {
-    return Attributes.builder()
-        .putAll(commonMetricDimensions)
-        .put(getDimensionName(HTTP_RESPONSE_STATUS_CODE), responseStatus.codeAsText().toString())
-        .put(
-            getDimensionName(HTTP_RESPONSE_STATUS_CODE_CATEGORY),
-            getVeniceHttpResponseStatusCodeCategory(responseStatus))
-        .put(getDimensionName(VENICE_RESPONSE_STATUS_CODE_CATEGORY), veniceResponseStatusCategory.getCategory())
-        .build();
-  }
-
   public void recordHealthyRequest(Double latency, HttpResponseStatus responseStatus, int keyNum) {
-    Attributes dimensions = getRequestMetricDimensions(responseStatus, VeniceResponseStatusCategory.SUCCESS);
-    healthyRequestMetric.record(1, dimensions);
-    keyCountMetric.record(keyNum, dimensions);
+    HttpResponseStatusCode httpResponseStatusCode = transformHttpResponseStatusToHttpResponseStatusCode(responseStatus);
+    HttpResponseStatusCodeCategory httpResponseStatusCodeCategory =
+        getVeniceHttpResponseStatusCodeCategory(responseStatus);
+    VeniceResponseStatusCategory veniceResponseStatusCategory = VeniceResponseStatusCategory.SUCCESS;
```

Comment:
The var name `veniceResponseStatusCategory` is too generic. Let's either rename it to just `success` or use `VeniceResponseStatusCategory.SUCCESS` directly in later function calls.

Reply:
These also had some duplicate code. Extracted it to a common method in the last diff.